core/src/gl/texture.cpp

```diff
@@ -155,18 +155,16 @@ void Texture::update(GLuint _textureUnit) {
 
         m_shouldResize = false;
 
-        m_subData.clear();
+    }
 
```

Comment:
The previous subdata should probably be cleared in the call to resize then since its stride will be wrong when the texture width has changed (even though the data will most likely be overwritten by following subdata)